File: WebDesk/Services/SettingsManager.cs
using System;
using System.IO;
using System.Text.Json;
using Forms = System.Windows.Forms;

namespace WebDesk.Services
{
    public class SettingsManager
    {
        private const string APP_NAME = "WebDesk";
        private const string REGISTRY_PATH = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Run";
        
        private readonly string _settingsPath;
        public string WallpaperPath { get; set; } = string.Empty;
        public bool AutoStart { get; set; } = true;

        public SettingsManager()
        {
            _settingsPath = Path.Combine(
                Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),
                APP_NAME,
                "settings.json"
            );
            Directory.CreateDirectory(Path.GetDirectoryName(_settingsPath)!);
            Load();
        }

        public void Save()
        {
            try
            {
                string json = JsonSerializer.Serialize(this);
                File.WriteAllText(_settingsPath, json);
                UpdateAutoStart();
            }
            catch (Exception ex)
            {
                Forms.MessageBox.Show($"Failed to save settings: {ex.Message}",
                    "Error", Forms.MessageBoxButtons.OK, Forms.MessageBoxIcon.Error);
            }
        }

        private void Load()
        {
            try
            {
                if (File.Exists(_settingsPath))
                {
                    string json = File.ReadAllText(_settingsPath);
                    var settings = JsonSerializer.Deserialize<SettingsManager>(json);
                    WallpaperPath = settings?.WallpaperPath ?? string.Empty;
                    AutoStart = settings?.AutoStart ?? true;
                }
            }
            catch (Exception ex)
            {
                Forms.MessageBox.Show($"Failed to load settings: {ex.Message}",
                    "Error", Forms.MessageBoxButtons.OK, Forms.MessageBoxIcon.Error);
            }
        }

        private void UpdateAutoStart()
        {
            Microsoft.Win32.RegistryKey? key = Microsoft.Win32.Registry.CurrentUser.OpenSubKey(
                REGISTRY_PATH,
                true);
                
            if (key != null)
            {
                if (AutoStart)
                {
                    key.SetValue(APP_NAME, System.Reflection.Assembly.GetExecutingAssembly().Location);
                }
                else
                {
                    key.DeleteValue(APP_NAME, false);
                }
            }
        }
    }
}

File: src/core/settings_manager.py
import json
import os
import winreg
from pathlib import Path

class SettingsManager:
    def __init__(self):
        self.app_name = "WebDesk"
        self.settings_path = Path(os.getenv('APPDATA')) / self.app_name / 'settings.json'
        self.wallpaper_path = ""
        self.auto_start = True
        self.load()
        
    def save(self):
        self.settings_path.parent.mkdir(parents=True, exist_ok=True)
        settings = {
            'wallpaper_path': self.wallpaper_path,
            'auto_start': self.auto_start
        }
        
        with open(self.settings_path, 'w') as f:
            json.dump(settings, f)
            
        self._update_autostart()
        
    def load(self):
        if self.settings_path.exists():
            with open(self.settings_path, 'r') as f:
                settings = json.load(f)
                self.wallpaper_path = settings.get('wallpaper_path', '')
                self.auto_start = settings.get('auto_start', True)
                
    def _update_autostart(self):
        key = winreg.OpenKey(
            winreg.HKEY_CURRENT_USER,
            r"SOFTWARE\Microsoft\Windows\CurrentVersion\Run",
            0, winreg.KEY_ALL_ACCESS
        )
        
        try:
            if self.auto_start:
                winreg.SetValueEx(
                    key,
                    self.app_name,
                    0,
                    winreg.REG_SZ,
                    sys.argv[0]
                )
            else:
                try:
                    winreg.DeleteValue(key, self.app_name)
                except WindowsError:
                    pass
        finally:
            key.Close()
